finetune/lm_base.py

```diff
@@ -367,6 +373,28 @@ def _define_placeholders(self):
         self.M = tf.placeholder(tf.float32, [None, self.max_length])  # sequence mask
         self.Y = tf.placeholder(tf.float32, [None, self.target_dim])  # classification targets
 
+    def lm_predict(self, max_length=None, seed_text=''):
+        """
+        Performs a prediction on the Language modeling objective given some seed text. It uses a noisy greedy decoding.
+        Temperature parameter for decoding is set in the config.
+        :param max_length: The maximum length to decode to.
+        :param seed_text: Defaults to the empty string. This will form the starting point to begin modelling
+        :return: A string containing the generated text.
+        """
+        seed_text_tokens = self.encoder.encode([seed_text])
+        self._build_model(n_updates_total=0, target_dim=self.target_dim, train=False)
+        string = [self.encoder['_start_']] + seed_text_tokens[0]
+        eos = [self.encoder['_classify_']]
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore")
+            for i in range(len(seed_text_tokens[0]), (max_length or self.max_length) - 1):
+                tokens, mask = self._array_format([string])
+                class_idx = self.sess.run(self.lm_predict_op, {self.X: tokens, self.M: mask})
+                string.append(class_idx[i])
+                if string[-1] == eos:
```

Comment:
I think this means that we will never early terminate.  It's a str and list comparison

Reply:
Nice catch, yeah I think you're right, I don't know where this came from, because it definitely was early terminating.